client/src/components/DiagnosticContainer.jsx

```diff
@@ -23,31 +24,49 @@ const dateOptions = {
   day: 'numeric',
 };
 
+// check if the survey has been submitted today
+const hasSubmitted = () => {
+  const date = window.localStorage.getItem('date');
+  const todaysDate = new Date().toISOString().slice(0, 10);
+  if (date === todaysDate) {
```

Comment:
Would there be a case where these two aren't equal?

Reply:
Yes. If I am logging in today today's date will be stored in the local storage, so when I login again tomorrow they will not be equals, so the function will return false and I will be shown the survey after logging in